ToolBox/Polygon/PolygonDownloaderProgram.cs

```diff
@@ -54,35 +58,50 @@ public static void PolygonDownloader(IList<string> tickers, string securityTypeS
-                var endDate = toDate.ConvertToUtc(TimeZones.NewYork);
 
+                // Get hours
                 var marketHoursDatabase = MarketHoursDatabase.FromDataFolder();
 
                 // Create an instance of the downloader
                 using (var downloader = new PolygonDataDownloader())
                 {
+                    // Server side parameters for multithreaded simultaneous API connections
+                    ServicePointManager.DefaultConnectionLimit = 4;
+
                     foreach (var ticker in tickers)
                     {
                         var symbol = Symbol.Create(ticker, securityType, market);
-
+                        
+                        // Get the correct time zones
                         var exchangeTimeZone = marketHoursDatabase.GetExchangeHours(market, symbol, securityType).TimeZone;
                         var dataTimeZone = marketHoursDatabase.GetDataTimeZone(market, symbol, securityType);
 
-                        foreach (var tickType in tickTypes)
+                        // Set the UTC times for correctness of the architecture
+                        var utcStartDate = fromDate.ConvertToUtc(dataTimeZone);  // TimeZones.NewYork);
+                        var utcEndDate = toDate.ConvertToUtc(dataTimeZone);  // TimeZones.NewYork);
+                        var numberOfDays = (utcEndDate - utcStartDate).TotalDays;
+
+                        // Iterate through each day, which is the Polygon.IO api architecture
+                        for (int i = 0; i <= numberOfDays; i++)
                         {
-                            // Download the data
-                            var data = downloader.Get(symbol, resolution, startDate, endDate, tickType)
-                                .Select(x =>
-                                    {
-                                        x.Time = x.Time.ConvertTo(exchangeTimeZone, dataTimeZone);
-                                        return x;
-                                    }
-                                );
+                            foreach (var tickType in tickTypes)
+                            {
+                                // Download the data
+                                var data = downloader.Get(symbol, resolution, utcStartDate.AddDays(i), utcEndDate.AddDays(i + 1), tickType)
+                                    .Select(x =>
+                                        {
+                                            x.Time = x.Time.ConvertTo(exchangeTimeZone, dataTimeZone);
+                                            return x;
+                                        }
+                                    );
 
-                            // Save the data
-                            var writer = new LeanDataWriter(resolution, symbol, dataDirectory, tickType);
-                            writer.Write(data);
+                                if ((data != null))
```

Comment:
data will never be null? might be empty though, suggest adding `ToList()` after the select and checking `count` here instead

Reply:
Done.

Reply:
`ToList()` should be applied in the response of `downloader.Get()` so we reuse it when calling `writer.Write(data);`, else the enumeration will be triggered more than once

Reply:
ok

Reply:
@Martin-Molinero     On days where this is no data returned from the API call, the returned response is a null value.   Are you wanting this NULL check done in a different part of the code ?